packages/presentation/lib/screen/home/home_page.dart

```diff
@@ -1,177 +1,157 @@
 import 'package:domain/usecase/check_number_usecase.dart';
 import 'package:domain/usecase/generate_number_usecase.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:presentation/screen/home/bloc/home_bloc.dart';
+import 'package:presentation/base/bloc_page.dart';
+import 'package:presentation/base/bloc_state.dart';
+import 'package:presentation/base/game_alert_dialog.dart';
+import 'package:presentation/screen/home/home_bloc.dart';
+import 'package:presentation/screen/home/home_state.dart';
 import 'package:presentation/screen/home/show_dialog_widget.dart';
 
 class MyApp extends StatelessWidget {
   const MyApp({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    return BlocProvider(
-      create: (context) => HomeBloc(
-        blocCheckUseCase: CheckNumberUseCase.instance,
-        blocGenerateUseCase: GenerateNumberUseCase.instance,
-      ),
-      child: MaterialApp(
-        debugShowCheckedModeBanner: false,
-        title: 'Generate',
-        theme: ThemeData(
-          primarySwatch: Colors.teal,
-        ),
-        home: const HomePage(title: 'Generate numbers'),
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      title: 'Generate',
+      theme: ThemeData(
+        primarySwatch: Colors.teal,
       ),
+      home: const HomePage(),
     );
   }
 }
 
-class HomePage extends StatefulWidget {
-  const HomePage({Key? key, required this.title}) : super(key: key);
-
-  final String title;
+class HomePage extends BlocScreen {
+  const HomePage({Key? key}) : super(key: key);
 
   @override
   State<HomePage> createState() => _HomePageState();
 }
 
-class _HomePageState extends State<HomePage> {
+class _HomePageState extends BlocScreenState<HomePage, HomeBloc> {
   late TextEditingController _controller;
 
+  _HomePageState()
+      : super(HomeBloc(GenerateNumberUseCase(), CheckNumberUseCase()));
+
   @override
   void initState() {
     super.initState();
     _controller = TextEditingController();
```

Comment:
you can create an instance of controller in bloc and use it there and dispose

Reply:
I prefer my way

Reply:
better to place it to bloc, because it will allow u to get/set the text to the widget